packages/keyring-api/src/api/asset.ts

```diff
@@ -0,0 +1,93 @@
+import type { Infer } from '@metamask/superstruct';
+import { literal, string } from '@metamask/superstruct';
+import {
+  CaipAssetIdStruct,
+  CaipAssetTypeStruct,
+  isPlainObject,
+} from '@metamask/utils';
+
+import { object, selectiveUnion } from '../superstruct';
+import { StringNumberStruct } from '../utils';
+
+/**
+ * This struct represents an asset. The `fungible` property is used to tag the
+ * union type and allow the following pattern:
+ *
+ * ```ts
+ * if (asset.fungible) {
+ *   // Use asset.type
+ * } else {
+ *   // Use asset.id
+ * }
+ * ```
+ * /**
```

Comment:
Is the `* /**` intended? Should this be `*` instead?

```suggestion
 *
```